Mathlib/Algebra/Ring/Ext.lean

```diff
@@ -0,0 +1,477 @@
+/-
+Copyright (c) 2023 Raghuram Sundararajan. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Raghuram Sundararajan
+-/
+import Mathlib.Algebra.Ring.Defs
+import Mathlib.Algebra.Group.Ext
+-- import Mathlib
+
+/-!
+# Extensionality lemmas for rings and similar structures
+
+In this file we prove extensionality lemmas for the ring-like structures defined in
+`Algebra/Ring/Defs`, ranging from `NonUnitalNonAssocSemiring` to `CommRing`. These extensionality
+lemmas take the form of asserting that two algebraic structures on a type are equal whenever the
+addition and multiplication defined by them are both the same.
+
+## Implementation details
+
+We follow `Algebra/Group/Ext` in using the term `(letI := i; HMul.hMul : R → R → R)` to refer to the
+multiplcation specified by a typeclass instance `i` on a type `R` (and similarly for addition). We
+abbreviate these using some local notations.
+
+Since `Algebra/Group/Ext` proved several injectivity lemmas, we do so as well — even if sometimes we
```

Comment:
```suggestion
Since `Mathlib/Algebra/Group/Ext` proved several injectivity lemmas, we do so as well — even if sometimes we
```

Reply:
Maybe with `.` instead of `/`?

Reply:
I think `Mathlib/Algebra/Group/Ext.lean` is probably the best bet, since that is easier to detect heuristically in doc-gen for linkification.

Reply:
IIRC I copied this style of referring to files from the commit conventions.

OTOH, I see the `.` style in the module header for `Algebra.Group.Defs` (without the `Mathlib` prefix), the `/` style (without `Mathlib` prefix, with `.lean` suffix) in `Algebra/Group/Basic.lean` and `Algebra/Group/UniqueProds.lean`, etc.

It's also noteworthy that the [contribution guidelines for documentation](https://leanprover-community.github.io/contribute/doc.html) mention that fully-qualified names (e.g., `Group.ext`) are turned into links to the online docs, so that could cause a conflict if `.`-style notation is used to refer to modules and there is a definition and a module with the same name eventually (although that's unlikely if the `Mathlib` prefix is used for the module names).

So which style should I go with?

Reply:
Let's use the `/` style for filenames. It's very easy to change that to dot notation later in a mathlib-wide sweep, but changing in the other direction is much harder.

Reply:
Done in 10abca9012.